Add install troubleshooting tips to Guides & Surveys SDK doc (#624)
amplitude/amplitude-docs: content/collections/guides_and_surveys/en/sdk.md

{{/partial:tabs}}
{{/partial:collapse}}

### Google Tag Manager

If you haven't already, first update to our latest template.You can find the update icon in the Templates page in GTM.

Next, on the Tags page, opt-in to Guides and Surveys via the Enable Guides and Surveys Plugin checkbox.

> ![](https://share.cleanshot.com/pKSGH94k)

Reviewer: I'm guessing it's order of operations, but I'll let @liuaud confirm

> ![](https://share.cleanshot.com/pKSGH94k)

Author: @markzegarelli I believe it's providing an order of operations, e.g. Step 1, Step 2, etc.

> ![](https://share.cleanshot.com/pKSGH94k)

Reviewer 2: @liuaud Can we get a version of it without the numbers? Since we don't discuss the steps in that order, it will be confusing to have them in there.

> ![](https://share.cleanshot.com/pKSGH94k)

Reviewer 2: @liuaud Just a clean screenshot, please. I'll highlight anything that needs to be highlighted, or describe as necessary.

If it has been installed, call `window.engagement._.user`. If that returns as `undefined`, then the plugin has not been set up properly.

You can further debug by calling `window.engagement._debugStatus()`. The output should look like this:

```json
{
    "user": {
        "user_id": "test-base-user-1vxxkg",
        "device_id": "62c5e45a-94ab-4090-b053-3f28e848763f",
        "user_properties": {
            "foo": "bar"
        }
    },
    "apiKey": "6ae8d3d7d48eadfb0b2489db692e14c9",
    "stateInitialized": true,
    "decideSuccessful": true,
    "num_guides_surveys": 2,
    "analyticsIntegrations": 1
}
```

Specifically, there should be a `user` object, `apiKey` needs to be set, `stateInitialized` needs to be true, `decideSuccessful` needs to be `true`, and there needs to be a non-zero `num_guides_surveys` if there are live guides or surveys in the config.

#### What do I do if my plugin hasn’t been set up properly?

If you are using our Browser SDK 2.0, first check the console for errors. If you don't see anything, verify that your code matches with the install instructions. In particular, double check that you have `amplitude.add(window.engagement.plugin())` in the code.

> If you see something like `amplitude is not defined` and `cannot read properties of undefined .add()`, this means that the G&S is trying to load before the amplitude SDK has been loaded. Please verify in your code that the Amplitude SDK is loaded before the G&S SDK. If you are having issues with this in Google Tag Manager, make sure that you have updated to our latest template.

Reviewer 2: ```suggestion
If you see something like `amplitude is not defined` and `cannot read properties of undefined .add()`, this means that the G&S is trying to load before the Amplitude SDK loads. Check your code to ensure that the Amplitude Browser SDK loads before the Guides and Surceys SDK. 

If you use Google Tag Manager, ensure you update to the latest Amplitude template.
```

> We do not support Amplitude's legacy Javascript SDK--please upgrade to the Browser SDK 2.0 and follow the install instructions above.

Reviewer 2: ```suggestion
Guides and Surveys requires Browser SDK 2.x.x or higher and doesn't support the legacy Amplitude JavaScript SDK.
```

@liuaud What's the minimum version of Browser SDK required here?

Author: Good question, @lchoward would you know the answer to this?

Author: @markzegarelli The G&S team do not know of a minimum version required, so I think we can just say Browser SDK 2.0?

#### Common root causes

##### `boot` being called multiple times

> This will result in unexpected behavior, especially for guides and surveys that should appear immediately.

Reviewer 2: ```suggestion
This results in unexpected behavior, especially for guides and surveys that should appear immediately.
```

{{partial:admonition type="info" heading=""}}

> If you are implementing G&S with `amplitude.add(window.engagement.plugin())`, you should not be calling `boot` at all. This is because the `add()` method will already boot with a very specific set of parameters.

Reviewer 2: ```suggestion
If you implement Guides and Sureys with `amplitude.add(window.engagement.plugin())`, don't call `boot`. The `add()` method includes this call with a very specific set of parameters.
```

##### Wrong project used

> Please make sure the API key belongs to the project that contains the guides and survey configurations and also belongs to the same project that the Analytics SDK is configured with.

Reviewer 2: ```suggestion
Ensure the API key you provide:

- is the same key you use to initialize the Browser SDK
- belongs to the project that contains the Guides and Surveys configuration
```

2. If `window.engagement` returns a valid response, enter `window.engagement._.user`. A return of `undefined` indicates an issue with the plugin configuration.

3. For additional debugging, enter `window.engagement._debugStatus()`. The output should be similar to:

```json
{
    "user": {
        "user_id": "test-base-user-1vxxkg",
        "device_id": "62c5e45a-94ab-4090-b053-3f28e848763f",
        "user_properties": {
            "foo": "bar"
        }
    },
    "apiKey": "6ae8d3d7d48eadfb0b2489db692e14c9",
    "stateInitialized": true,
    "decideSuccessful": true,
    "num_guides_surveys": 2,
    "analyticsIntegrations": 1
}
```

Verify that:
- the `user` object is present
- `apiKey` is set
- `stateInitialized` is `true`
- `decideSuccessful` is `true`
- `num_guides_surveys` is a non-zero integer if a guide or survey should be display on the page.

#### Verify plugin configuration

If you use Amplitude Browser SDK 2.0, check the browser's console for errors. If there are none, verify that your code matches code provided in the installation instructions. In particular, ensure that  `amplitude.add(window.engagement.plugin())` is present in the code.

> If you see something like `amplitude is not defined` and `cannot read properties of undefined .add()`, this means that the G&S is trying to load before the Amplitude SDK loads. Check your code to ensure that the Amplitude Browser SDK loads before the Guides and Surceys SDK. 

Reviewer 2: ```suggestion
If you see something like `amplitude is not defined` and `cannot read properties of undefined .add()`, this means that the G&S is trying to load before the Amplitude SDK loads. Check your code to ensure that the Amplitude Browser SDK loads before the Guides and Surveys SDK. 
```

If you use Google Tag Manager, ensure you update to the latest Amplitude template.

We do not support Amplitude's legacy Javascript SDK--please upgrade to the Browser SDK 2.0 and follow the install instructions above.

#### Common root causes

##### `boot` being called multiple times

This results in unexpected behavior, especially for guides and surveys that should appear immediately.

{{partial:admonition type="info" heading=""}}

> If you implement Guides and Sureys with `amplitude.add(window.engagement.plugin())`, don't call `boot`. The `add()` method includes this call with a very specific set of parameters.

Reviewer 2: ```suggestion
If you implement Guides and Surveys with `amplitude.add(window.engagement.plugin())`, don't call `boot`. The `add()` method includes this call with a very specific set of parameters.
```
